src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -258,6 +258,14 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         // add a query string if needed
         $extra = array_diff_key($parameters, $variables, $defaults);
+        foreach($parameters as $key => $value) {
+            if (isset($defaults[$key]) &&
+                !array_key_exists($key, $variables) &&
+                $defaults[$key] !== $value
+            ) {
```

Comment:
Can you put the condition in a single line please? It would be more consistent with rest of the code base.